### DISCUSSION TITLE
Mixing Flatcar with other distros in a K8S cluster?

### DISCUSSION THREAD

**Discussion by abceleung** [flatcar/Flatcar#1555] [Q&A] [answered]
(Newbie question, never setup a K8S cluster before)

I have two x86 machines and a Raspberry Pi 5. I was planning to install Talos on all of them, but I just found out that Raspberry Pi 5 is not supported by almost all distros, including Talos, FCOS and Flatcar.

I think I have to install Raspbian on the Pi and setup K8S manually. Would the K8S setup still work if I install Flatcar on the other two x86 machines?

(FWIW, this is the [answer from a maintainer of Talos](https://github.com/siderolabs/talos/discussions/9435) when I asked them a similar question)

**Comment by tormath1** [comment, marked as answer]
Hi @abceleung sorry for the delay. Flatcar uses vanilla Kubernetes and has no strict requirements regarding the used control-plane. I think you should be able to mix Flatcar nodes with other distro in your Kubernetes cluster. 

Let us know how it goes, that might be interesting to document.

**Reply by seeinlab** [reply]
That's great, I was looking for this because running Flatcar directly on a Pi5 isn't currently feasible (due to UEFI support on the Pi5).

I came here because I read about running worker nodes (EKS) using Flatcar. I want to run my homelab using Flatcar instead of Talos; simply put, instead of "marketing terms," ​​the Flatcar documentation has more real-world use cases.

**Reply by tormath1** [reply]
Thanks and for the record since the last message on 2024 on this discussion, I ran a kubernetes cluster mixing Flatcar and Fedora CoreOS nodes. :)

**Reply by seeinlab** [reply]
> Thanks and for the record since the last message on 2024 on this discussion, I ran a kubernetes cluster mixing Flatcar and Fedora CoreOS nodes. :)

glad to hear that. i will try it
